src/cdk/v2/destinations/bluecore/utils.js

```diff
@@ -169,6 +204,45 @@ const createProductForStandardEcommEvent = (message, eventName) => {
   }
   return null;
 };
+/**
+ * Function: populateAccurateDistinctId
+ *
+ * Description:
+ * This function is used to populate the accurate distinct ID based on the given payload and message.
+ *
+ * Parameters:
+ * - payload (object): The payload object containing the event and other data.
+ * - message (object): The message object containing the user data.
+ *
+ * Returns:
+ * - distinctId (string): The accurate distinct ID based on the given payload and message.
+ *
+ * Throws:
+ * - InstrumentationError: If the distinct ID could not be set.
+ *
+ */
+const populateAccurateDistinctId = (payload, message) => {
+  let distinctId;
+  const bluecoreExternalId = getDestinationExternalID(message, 'bluecoreExternalId');
+  if (!isDefinedAndNotNullAndNotEmpty(bluecoreExternalId)) {
+    if (payload.event === 'identify') {
+      distinctId = getFieldValueFromMessage(message, 'userId');
+    } else {
+      // email is always a more preferred distinct_id
```

Comment:
Any particular reason why email is more preferred in case of track but not identify?
Also would this result is user flow to be same or can it result in users and their activity to not merge ?